src/controllers/tripsController.js

```diff
@@ -144,6 +151,100 @@ class tripsController {
       return errorResponse(res, error.status || 500, error.message);
     }
   }
+
+  static async searchTrips(req, res) {
+    const {
+      body: {
+        searchKey
+      },
+      query: {
+        page, limit,
+      },
+      user: {
+        userID, role, email
+      }
+    } = req;
+    const pagination = paginate(page, limit);
+    try {
+      let found;
+      const searchQuery = [
+        where(
+          cast(col('travelRequests.requestId'), 'varchar'),
+          { [Op.like]: `%${searchKey}%` }
+        ),
+        where(
+          cast(col('travelRequests.type'), 'varchar'),
+          { [Op.like]: `%${searchKey}%` }
+        ),
+        where(
+          cast(col('travelRequests.gender'), 'varchar'),
+          { [Op.like]: `%${searchKey}%` }
+        ),
+        where(
+          cast(col('travelRequests.status'), 'varchar'),
+          { [Op.like]: `%${searchKey}%` }
+        ),
+        where(
+          cast(col('travelRequests.stops'), 'varchar'),
+          { [Op.like]: `%${searchKey}%` }
+        ),
+
+        where(
+          cast(col('travelRequests.status'), 'varchar'),
+          { [Op.like]: `%${searchKey}%` }
+        ),
+        { from: { [Op.like]: `%${searchKey}%`, }, },
+        { to: { [Op.like]: `%${searchKey}%`, }, },
+        { from: { [Op.like]: `%${searchKey}%`, }, },
+        { to: { [Op.like]: `%${searchKey}%`, }, },
+        { reason: { [Op.like]: `%${searchKey}%`, }, },
+      ];
+      if (role === 'requester') {
+        found = await travelRequests.findAll({
+          where: {
+            userID,
+            [Op.or]: searchQuery
+          },
+          attributes: {
+            exclude: ['id', 'userID', 'passportNumber', 'gender', 'createdAt', 'updatedAt']
+          },
+          offset: pagination.offset,
+          limit: pagination.limit,
+        });
+      }
+      if (role === 'manager') {
+        found = await travelRequests.findAll({
+          where: {
+            manager: email,
+            [Op.or]: searchQuery
+          },
+          attributes: {
+            exclude: ['id', 'userID', 'passportNumber', 'gender', 'createdAt', 'updatedAt']
+          },
+          offset: pagination.offset,
+          limit: pagination.limit,
+        });
+      }
+      found = await travelRequests.findAll({
+        where: {
+          [Op.or]: searchQuery
+        },
+        attributes: {
+          exclude: ['id', 'userID', 'passportNumber', 'gender', 'createdAt', 'updatedAt']
+        },
+        offset: pagination.offset,
+        limit: pagination.limit,
+      });
+      return res.status(200).json({
+        status: 200,
+        message: req.i18n.__('retrievedSuccessfully'),
```

Comment:
```suggestion
        message: setLanguage(req.user.preferedLang).__('retrievedSuccessfully'),
```

@blaise82 Would you please make use of ```setLanguage``